contracts/connection-router/src/state/payload_hex.rs

```diff
@@ -0,0 +1,19 @@
+use cosmwasm_std::HexBinary;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+pub fn serialize<S>(value: &[u8; 32], serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    HexBinary::from(value).serialize(serializer)
+}
+
+pub fn deserialize<'de, D>(deserializer: D) -> Result<[u8; 32], D::Error>
+where
+    D: Deserializer<'de>,
+{
+    HexBinary::deserialize(deserializer)?
+        .to_array::<32>()
+        .map_err(|err| Error::custom(err))
+}
```

Comment:
Why not?
```suggestion
pub fn serialize<S, const N: usize>(value: &[u8; N], serializer: S) -> Result<S::Ok, S::Error>
where
    S: Serializer,
{
    HexBinary::from(value).serialize(serializer)
}

pub fn deserialize<'de, D, const N: usize>(deserializer: D) -> Result<[u8; N], D::Error>
where
    D: Deserializer<'de>,
{
    HexBinary::deserialize(deserializer)?
        .to_array::<N>()
        .map_err(|err| Error::custom(err))
}
```

Reply:
Also, I am not a fan of this mod name. It's really just serializing `[u8: N]` as hex string. Shouldn't have to be tied with payload_hash.

Reply:
nice! I wasn't aware you could do `const N`. The reason I went with `payload_hash` is that I had it as `hex` first, but that collides with the std libs hex crate and I couldn't figure out how to resolve that naming conflict. I'm open for suggestions